src/middlewared/middlewared/plugins/activedirectory.py

```diff
@@ -849,31 +842,11 @@ class Config:
 
     @private
     async def _set_state(self, state):
-        await self.middleware.call('cache.put', 'AD_State', state.name)
+        return await self.middleware.call('directoryservices.set_state', {'activedirectory': state.name})
 
     @accepts()
     async def get_state(self):
```

Comment:
I think we need either to make this private or add a doc stating its a shortcut to directoryservices.get_state for active directory

Ditto for nis/ldap